apps/condo/domains/notification/transports/push.js

```diff
@@ -146,6 +150,28 @@ async function send ({ notification, data, user, remoteClient } = {}, isVoIP = f
             const adapter = ADAPTERS[transport]
             const payload = { tokens, pushTypes, appIds, notification, data }
             const [isOk, result] = await adapter.sendNotification(payload, isVoIP)
+            if (adapter === PUSH_TRANSPORT_FIREBASE) {
+                // handling expired token error. https://firebase.google.com/docs/cloud-messaging/manage-tokens?hl=ru#detect-invalid-token-responses-from-the-fcm-backend
+                if (get(result, 'responses')) {
+                    for (const res of result.responses) {
+                        const context = getSchemaCtx('RemoteClient')
+                        if (get(res, 'error.code') === 'messaging/registration-token-not-registered') {
+                            const field = isVoIP ? 'pushTokenVoIP' : 'pushToken'
+                            const [remoteClient] = await find('RemoteClient', {
+                                [field]: res.pushToken,
+                                deletedAt: null,
+                            })
+
+                            !!remoteClient && await RemoteClient.update(context, get(remoteClient, 'id'), {
+                                [field]: null,
+                                dv: 1,
+                                sender: { dv: 1, fingerprint: 'internal-update_token-not-registered' },
+                            })
+                            logger.info({ msg: 'Remove expired FCM token', remoteClientId: remoteClient.id, field })
```

Comment:
Why is the logger not in the condition `if (remoteClient) { RemoteClient.update and logger.info }`?